test/unit/lib/plugins/aws/deployFunction.test.js

```diff
@@ -591,4 +367,412 @@ describe('test/unit/lib/plugins/aws/deployFunction.test.js', () => {
+              PackageType: 'Zip',
+            },
+          },
+        },
+      },
+      configExt: {
+        provider: {
+          environment: {
+            ANOTHERVAR: 'anothervalue',
+          },
+        },
+        functions: {
+          foo: {
+            awsKmsKeyArn: kmsKeyArn,
+            description,
+            handler,
+            environment: {
+              VARIABLE: 'value',
+            },
+            name: functionName,
+            memorySize,
+            onError: onErrorHandler,
+            role,
+            timeout,
+            vpc: {
+              securityGroupIds: ['sg-111', 'sg-222'],
+              subnetIds: ['subnet-111', 'subnet-222'],
+            },
+            layers: [layerArn, secondLayerArn],
+          },
+        },
+      },
+    });
+
+    expect(updateFunctionConfigurationStub).to.be.calledWithExactly({
+      FunctionName: functionName,
+      KMSKeyArn: kmsKeyArn,
+      Description: description,
+      Handler: handler,
+      Environment: {
+        Variables: {
+          ANOTHERVAR: 'anothervalue',
+          VARIABLE: 'value',
+        },
+      },
+      MemorySize: memorySize,
+      Timeout: timeout,
+      DeadLetterConfig: {
+        TargetArn: onErrorHandler,
+      },
+      Role: role,
+      VpcConfig: {
+        SecurityGroupIds: ['sg-111', 'sg-222'],
+        SubnetIds: ['subnet-111', 'subnet-222'],
+      },
+      Layers: [layerArn, secondLayerArn],
+    });
+    expect(stdoutData).to.include('Successfully updated function');
+  });
+
+  it('should skip updating properties that contain references', async () => {
+    const { stdoutData } = await runServerless({
+      fixture: 'function',
+      cliArgs: ['deploy', 'function', '-f', 'foo'],
+      awsRequestStubMap: {
+        ...awsRequestStubMap,
+        Lambda: {
+          ...awsRequestStubMap.Lambda,
+          getFunction: {
+            Configuration: {
+              LastModified: '2020-05-20T15:34:16.494+0000',
+              PackageType: 'Zip',
+            },
+          },
+        },
+      },
+      configExt: {
+        functions: {
+          foo: {
+            name: functionName,
+            role,
+            timeout,
+            vpc: {
+              securityGroupIds: ['sg-111', { Ref: 'mySGRef' }],
+              subnetIds: ['subnet-111', 'subnet-222'],
+            },
+            environment: {
+              VARIABLE: {
+                Ref: 'SomeReference',
+              },
+            },
+          },
+        },
+      },
+    });
+
+    expect(updateFunctionConfigurationStub).to.be.calledWithExactly({
+      FunctionName: functionName,
+      Handler: 'index.handler',
+      Timeout: timeout,
+      VpcConfig: {
+        SubnetIds: ['subnet-111', 'subnet-222'],
+      },
+      Role: role,
+    });
+    expect(stdoutData).to.include('Successfully updated function');
+  });
+
+  it('should cast non-string env variables to strings during update', async () => {
```

Comment:
Let's skip this test. String coercion is done at configuration validation step. This functionality in all cases receives property values coerced to defined types